src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extension.cs

```diff
@@ -1132,6 +1112,67 @@ internal sealed override ParameterSymbol? ExtensionParameter
             }
         }
 
+        public string GetExtensionGroupingMetadataName()
+        {
+            Debug.Assert(IsExtension);
+
+            if (_lazyExtensionInfo is null)
+            {
+                Interlocked.CompareExchange(ref _lazyExtensionInfo, new ExtensionInfo(), null);
+            }
+
+            if (_lazyExtensionInfo.LazyExtensionGroupingName is null)
+            {
+                _lazyExtensionInfo.LazyExtensionGroupingName = WellKnownMemberNames.ExtensionMarkerMethodName + RawNameToHashString(ComputeExtensionGroupingRawName());
+            }
+
+            return _lazyExtensionInfo.LazyExtensionGroupingName;
+        }
+
+        public string GetExtensionMarkerMetadataName()
+        {
+            Debug.Assert(IsExtension);
+
+            if (_lazyExtensionInfo is null)
+            {
+                Interlocked.CompareExchange(ref _lazyExtensionInfo, new ExtensionInfo(), null);
+            }
+
+            if (_lazyExtensionInfo.LazyExtensionMarkerName is null)
+            {
+                _lazyExtensionInfo.LazyExtensionMarkerName = "<Marker>$" + RawNameToHashString(ComputeExtensionMarkerRawName()); // PROTOTYPE: Add a constant for this prefix.
```

Comment:
If there are any PROTOTYPE comments in this PR that you don't consider blocking for merging the feature branch back to `main`, it would be helpful to use linked issues instead or tagging them somehow.